bgpd/bgp_fsm.c

```diff
@@ -2974,10 +2975,10 @@ unsigned int bgp_peer_gr_action(struct peer *peer, int old_peer_state,
 
 		BGP_PEER_GR_GLOBAL_INHERIT_UNSET(peer);
 
-		if (new_peer_state == bgp_gr_global_mode) {
-			/*This is incremental updates i.e no tear down
-			 *of the existing session
-			 *as the peer is already working in the same mode.
+		if ((int)new_peer_state == (int)bgp_gr_global_mode) {
```

Comment:
both are enum_peer_mode right?  Why do we need casting to (int)?

Reply:
not really, they are different 🤷